eng/scripts/prepare-publish.js

```diff
@@ -19,45 +37,50 @@ const Major = 3;
  * Set this to false to test this script without creating a new branch, checking for changes etc.
  * DO NOT LEAVE TO FALSE
  */
-const production = true;
+const production = !args.values.noCommit;
 
 let branch;
 
 if (production) {
   // Create and checkout branches
   branch = `publish/${Date.now().toString(36)}`;
-  doubleRun("git", "checkout", "-b", branch);
+  log("Creating branch in both repos", branch);
+  await doubleRun("git", "checkout", "-b", branch);
 }
 
 // Check that we have a clean slate before starting
 if (production) {
-  checkPrePublishState();
+  await checkPrePublishState();
 }
 
 // Update the typespec core submodule
+log("Updating typespec core submodule");
 typespecRun("git", "fetch", "https://github.com/microsoft/typespec", "main");
 if (production) {
   typespecRun("git", "merge", "--ff-only", "FETCH_HEAD");
 }
 // Stage the typespec core publish
 typespecRun("pnpm", "change", "version");
-typespecRun("pnpm", "update-latest-docs");
+if (!args.values.onlyBumpVersions) {
+  typespecRun("pnpm", "update-latest-docs");
+}
```

Comment:
Shouldn't the 'git add -A' below only occur if there are changes (i.e. with the 'git commit -m' on line 51)